### PR TITLE
Add support for "modular" schemas

### PR DESCRIPTION
**Pull request by joshbode**
- `components/schemas` references of the form `foo.bar.thing` are collated into a "modular" structure:
```
models/
  __init__.py
  foo/
    __init__.py
    bar.py  <- Thing model in here
```
- `UUID` changed from `pydantic.UUID` (doesn't exist) to `uuid.UUID`

### CODE REVIEW

File: datamodel_code_generator/parser/openapi.py [hunk 1]
```diff
@@ -99,42 +111,50 @@ def parse_all_of(self, name: str, obj: JsonSchemaObject) -> List[DataType]:
...
 
     def parse_object_fields(self, obj: JsonSchemaObject) -> List[DataModelField]:
-        requires: Set[str] = set(obj.required or [])
+        properties: Dict[str, JsonSchemaObject] = (
+            obj.properties if obj.properties is not None else {}
+        )
+        requires: Set[str] = {*obj.required} if obj.required is not None else {*()}
         fields: List[DataModelField] = []
 
-        for field_name, filed in obj.properties.items():  # type: ignore
```
Comment (koxudaxi) [inline datamodel_code_generator/parser/openapi.py:105]: It's a typo. good catch :)

File: tests/parser/test_openapi.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def render(self) -> str:
         ('string', 'password', 'SecretStr', 'pydantic', 'SecretStr'),
         ('string', 'email', 'EmailStr', 'pydantic', 'EmailStr'),
         ('string', 'uri', 'UrlStr', 'pydantic', 'UrlStr'),
-        ('string', 'uuid', 'UUID', 'pydantic', 'UUID'),
+        ('string', 'uuid', 'UUID', 'uuid', 'UUID'),
```
Comment (koxudaxi) [inline tests/parser/test_openapi.py]: I mistake. thank you

File: datamodel_code_generator/parser/openapi.py [hunk 3]
```diff
@@ -99,42 +111,50 @@ def parse_all_of(self, name: str, obj: JsonSchemaObject) -> List[DataType]:
...
+                if field.properties:
+                    class_name = self.get_class_name(field_name)
+                    self.parse_object(class_name, field)
+                    field_types = [
+                        self.data_type(
+                            type=class_name, ref=True, version_compatible=True
+                        )
+                    ]
+                else:
+                    field_types = [self.data_type(type="Dict[str, Any]")]
```
Comment (koxudaxi) [inline datamodel_code_generator/parser/openapi.py:146]: great!! I didn't expect this case.

**Comment by koxudaxi** [comment]
Thank you for creating this PR.
Great work. All changed is very nice.

> components/schemas references of the form foo.bar.thing are collated into a "modular"

I don't know this idea. I will try it.

Would you please cover all codes by the unittest?
I want to merge the PR when coverage is 100%.

**Comment by prostomarkeloff** [comment]
LGTM

**Comment by joshbode** [comment]
Will do - the code to write out to the module directory wasn't tested (apart from manually), so I'll work on that today :)

**Comment by joshbode** [comment]
OK - test coverage is 100% again :)

**Comment by koxudaxi** [comment]
Great work!!
Thank you very much.
I will release tonight :)

**Comment by joshbode** [comment]
Great! Thank you for reviewing and accepting it

**Comment by koxudaxi** [comment]
I found a changed in a single quote to double quote. I will fix the change before release.

**Comment by koxudaxi** [comment]
@joshbode 
I have released version 0.2.5 
Thank you very much.

I will be waiting next time 😄

**Pull request merged**